src/main/config/config.c

```diff
@@ -415,98 +415,20 @@ static void validateAndFixConfig(void)
         gyroConfigMutable()->gyro_hardware_lpf = GYRO_HARDWARE_LPF_NORMAL;
     }
 
-// clear features that are not supported.
-// I have kept them all here in one place, some could be moved to sections of code above.
-
-#ifndef USE_RX_PPM
-    featureDisableImmediate(FEATURE_RX_PPM);
-#endif
-
-#ifndef USE_SERIALRX
-    featureDisableImmediate(FEATURE_RX_SERIAL);
-#endif
-
-#if !defined(USE_SOFTSERIAL)
-    featureDisableImmediate(FEATURE_SOFTSERIAL);
-#endif
-
-#ifndef USE_RANGEFINDER
-    featureDisableImmediate(FEATURE_RANGEFINDER);
-#endif
-
-#ifndef USE_TELEMETRY
-    featureDisableImmediate(FEATURE_TELEMETRY);
-#endif
-
-#ifndef USE_RX_PWM
-    featureDisableImmediate(FEATURE_RX_PARALLEL_PWM);
-#endif
-
-#ifndef USE_RX_MSP
-    featureDisableImmediate(FEATURE_RX_MSP);
-#endif
-
-#ifndef USE_LED_STRIP
-    featureDisableImmediate(FEATURE_LED_STRIP);
-#endif
-
-#ifndef USE_DASHBOARD
-    featureDisableImmediate(FEATURE_DASHBOARD);
-#endif
-
-#ifndef USE_OSD
-    featureDisableImmediate(FEATURE_OSD);
-#endif
-
-#ifndef USE_SERVOS
-    featureDisableImmediate(FEATURE_SERVO_TILT | FEATURE_CHANNEL_FORWARDING);
-#endif
-
-#ifndef USE_TRANSPONDER
-    featureDisableImmediate(FEATURE_TRANSPONDER);
-#endif
-
-#ifndef USE_RX_SPI
-    featureDisableImmediate(FEATURE_RX_SPI);
-#endif
-
-#ifndef USE_ESC_SENSOR
-    featureDisableImmediate(FEATURE_ESC_SENSOR);
-#endif
-
-#if !defined(USE_ADC)
-    featureDisableImmediate(FEATURE_RSSI_ADC);
-#endif
-
-if (systemConfig()->configurationState == CONFIGURATION_STATE_UNCONFIGURED) {
-
-#ifdef USE_DASHBOARD
-    featureEnableImmediate(FEATURE_DASHBOARD);
-#endif
-#ifdef USE_LED_STRIP
-    featureEnableImmediate(FEATURE_LED_STRIP);
-#endif
-#ifdef USE_OSD
-    featureEnableImmediate(FEATURE_OSD);
-#endif
-#ifdef USE_RANGEFINDER
-    featureEnableImmediate(FEATURE_RANGEFINDER);
-#endif
-#ifdef USE_SERVOS
-    featureEnableImmediate(FEATURE_CHANNEL_FORWARDING);
-    featureEnableImmediate(FEATURE_SERVO_TILT);
-#endif
+    // clear features that are not supported.
+    featureDisableImmediate(~featuresSupportedByBuild);
+
+    if (systemConfig()->configurationState == CONFIGURATION_STATE_UNCONFIGURED) {
+        // enable some compiled-in features by default
+        uint32_t autoFeatures =
+            FEATURE_DASHBOARD | FEATURE_LED_STRIP | FEATURE_OSD | FEATURE_RANGEFINDER
+            | FEATURE_CHANNEL_FORWARDING | FEATURE_SERVO_TILT
```

Comment:
Think these are most common, the other ones should be enabled manually

```suggestion
            FEATURE_LED_STRIP | FEATURE_OSD
```

Reply:
@haslinghuis : I tried to avoid changing behavior of code where not necessary. We can easily change defaults in separate PR, probably together with target defaults